app/routes/task_routes.py

```diff
@@ -1 +1,151 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, abort, make_response, request, Response
+from ..db import db
+from ..models.task import Task
+from .route_utilities import validate_model
+import json
+from datetime import datetime, timezone
+import requests
+import os
+
+tasks_bp = Blueprint("tasks_bp", __name__, url_prefix="/tasks")
+
+
+@tasks_bp.post("")
+def create_task():
+
+    request_body = request.get_json()
+
+    try:
+        new_task = Task.from_dict(request_body)
+    except KeyError as e:
+        if e.args[0] == "title" or e.args[0] == "description":
+            response = {"details": "Invalid data"}
+        return make_response(response, 400)
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    response = {"task": new_task.create_response()}
+
+    return response, 201
+
+
+@tasks_bp.get("")
+def get_all_tasks():
+    query = db.select(Task)
+    title_param = request.args.get("title")
+
+    # if title_param:
+    #     query = query.where(Task.title == title_param)
+
+    # description_param = request.args.get("description")
+    # if description_param:
+    #     query = query.where(Task.description.ilike(f"%{description_param}%"))
+
+    # completed_at_param = request.args.get("completed_at")
+    # if completed_at_param:
+    #     query = query.where(Task.completed_at.ilike(f"%{completed_at_param}%"))
+
+    sort_param = request.args.get("sort")
+    if sort_param == "asc":
+        query = query.order_by(Task.title.asc())
+    elif sort_param == "desc":
+        query = query.order_by(Task.title.desc())
+    else:
+        query = query.order_by(Task.id)
+
+    tasks = db.session.scalars(query).all()
+
+    response = [task.create_response() for task in tasks]
+
+    return response
+
+
+@tasks_bp.get("/<task_id>")
+def get_one_task(task_id):
+    try:
+        task = validate_model(Task, task_id)
+    except:
+        return make_response({"details": f"Task {task_id} not found"}, 404)
+
+    response = {"task": task.create_response()}
+
+    return response
```

Comment:
See comment above in `goal_routes.py` for `get_one_goal`. I'd prefer the error handling to happen in `validate_model` so that this route can be more concise and only be concerned with one responsibility of getting a task (not having to handle errors too).

When `validate_model` is refactored, then this route would be just two lines

```suggestion
def get_one_task(task_id):
        task = validate_model(Task, task_id)
        return {"task": task.create_response()}
```